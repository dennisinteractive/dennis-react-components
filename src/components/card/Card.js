import React from "react";

const Card = ({
  title,
  path,
  image
}) => (
  <a href={`/${path}`} className='card'>
    <h2>{title}</h2>
    <img src={image}/>
  </a>
)

export default Card