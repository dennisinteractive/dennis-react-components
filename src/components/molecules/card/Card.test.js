import React from 'react';
import Card from './Card';

describe('<Card>', () => {
  let wrapper;
  const defProps = {
    title: 'Test Title',
    path: 'test-path',
    image: {
      src: 'https://placeimg.com/640/480/animals',
      alt: 'Test Image'
    },
  };

  beforeEach(() => {
    wrapper = shallow(
      <Card {...defProps} />,
    );
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
