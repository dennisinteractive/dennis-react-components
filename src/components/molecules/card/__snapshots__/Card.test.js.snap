// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card> should match snapshot 1`] = `
<ForwardRef
  className="card"
  href="test-path"
>
  <ForwardRef
    className="title"
    size={3}
    text="Test Title"
  />
  <ForwardRef
    alt="Test Image"
    src="https://placeimg.com/640/480/animals"
  />
</ForwardRef>
`;
